### DISCUSSION TITLE
What was the intention to use mpsc::channel in EqualReader?

### DISCUSSION THREAD

**Discussion by kolbma** [tiny-http/tiny-http#255] [Q&A]
There is an empty 10 years old _TODO_ at

https://github.com/tiny-http/tiny-http/blob/e2215636a73c7c3ac2d0a59d607454591f2cb2ac/src/request.rs#L214-L217

`EqualReader` returns in the tuple an unused `Receiver` of a `mpsc::channel`.  
To [fix](https://github.com/kolbma/tiny-http/commit/ccdd84ca82e0beb74390831dcf7ba4c43f54459c) an OOM as a side-effect I've changed the API here to provide the `Option<Sender>` as parameter and using None.    
This is the only usage of `EqualReader` in the project.

But maybe @tomaka or anyone else can remember what has been the intention to start thinking about using _channel_-communication in the drop of `EqualReader` and what maybe could be put after the _TODO_-colon?

**Comment by tomaka** [comment]
I don't remember, sorry
